### PR TITLE
[DR-2560] Add properties field to snapshot

### PR DESCRIPTION
**Pull request by samanehsan**
- Add `properties` JSONB field to snapshot table
- Update the snapshot create, retrieve and patch endpoint to include the new field

### CODE REVIEW

File: src/main/java/bio/terra/service/snapshot/SnapshotDao.java [hunk 1]
```diff
@@ -169,8 +169,8 @@ public void createAndLock(Snapshot snapshot, String flightId) {
     logger.debug("createAndLock snapshot " + snapshot.getName());
 
     String sql =
-        "INSERT INTO snapshot (name, description, profile_id, project_resource_id, id, consent_code, flightid, creation_information) "
-            + "VALUES (:name, :description, :profile_id, :project_resource_id, :id, :consent_code, :flightid, :creation_information::jsonb) ";
+        "INSERT INTO snapshot (name, description, profile_id, project_resource_id, id, consent_code, flightid, creation_information, properties) "
+            + "VALUES (:name, :description, :profile_id, :project_resource_id, :id, :consent_code, :flightid, :creation_information::jsonb, cast(:properties as jsonb)) ";
```
Comment (nmalfroy) [inline src/main/java/bio/terra/service/snapshot/SnapshotDao.java]: am I reading this right that creation_information and properties are both jsonb but accessed differently?  I'm curious why you're using different methods

**Review by nmalfroy — APPROVED** [review]
Looks great!  Just had a question about the different interactions with `jsonb` columns

File: src/main/java/bio/terra/service/snapshot/SnapshotDao.java (hunk 1 above)
Comment (samanehsan) [reply src/main/java/bio/terra/service/snapshot/SnapshotDao.java]: Good point! I made those consistent

**Review by okotsopoulos — APPROVED** [review]
Looks good to me (and speedy development too, glad that the earlier work on dataset properties gave an opportunity to work out the kinks with jsonb!)

**Pull request merged**